snapshot-labs/sx-ui
# feat: display proposal state

### Summary

Closes: https://github.com/snapshot-labs/pitches/issues/51

Changes:
- Until we get real integration of Central icon System it's just embedding SVGs like this. I didn't want to add those as separate icons yet.
- Not showing `#PROPOSAL_NUMBER` for read only spaces as they don't have numerical IDs.
- Pending uses box with minus sign, active uses empty box, passed/executed uses box with plus sign, rejected uses box with cross sign (not all were defined).
- Replaced `has_started`, `executed`, `has_ended` flags for unified `state` prop.

### How to test

1. Go to few spaces, for example
    1. http://localhost:8080/#/s-tn:radiantcapital.eth
    2. http://localhost:8080/#/s:arbitrumfoundation.eth
    3. http://localhost:8080/#/sep:0xea4322cc9be9ae0c9400984209c858031b1c4438
2. Overview only shows single list of proposals.
3. You can see all proposal types (pending/active/passed/rejected/executed)

### Screenshots

<img width="1106" alt="image" src="https://github.com/snapshot-labs/sx-ui/assets/1968722/5f300291-92c2-4539-a2a6-4a3d7484c478">

<img width="1105" alt="image" src="https://github.com/snapshot-labs/sx-ui/assets/1968722/1a1ebf79-aa0e-42ea-90a2-e3367b27453d">
